### PR TITLE
sourcify lib: verify from etherscan

### PR DESCRIPTION
**Pull request by rimrakhimov**
Add a new endpoint: `verify_from_sourcify` to sourcify client. Updates the error handling mechanisms

### CODE REVIEW

File: libs/sourcify/src/client.rs [hunk 1]
```diff
@@ -94,12 +92,50 @@ impl Client {
...
+        #[serde(rename_all = "camelCase")]
+        struct Request<'a> {
+            chain_id: &'a str,
+            address: String,
+        }
+
+        let request = Request {
+            chain_id,
+            address: contract_address.to_string(),
+        };
```
Comment (sevenzing) [inline libs/sourcify/src/client.rs:114]: I think this way is okay, but I just wonder why not simply use serde_json::json
```
let request = serde_json::json!({
    "chainId": chain_id,
    "address": contract_address.
});
```

**Review by sevenzing — APPROVED** [review]

File: libs/sourcify/src/client.rs (hunk 1 above)
Comment (rimrakhimov) [reply libs/sourcify/src/client.rs:114]: I believe it may done both ways, the separate structure was just the first which came up in my head

**Comment by sevenzing** [comment]
maybe rename this crate to `sourcify-client` ?

**Comment by rimrakhimov** [comment]
> maybe rename this crate to sourcify-client ?

We may discuss that in a separate thread I believe

**Pull request merged**